atl-checker/src/algorithms/certain_zero/search_strategy/linear_optimize.rs

```diff
@@ -0,0 +1,515 @@
+                            // Find distance from the current value, to the solution
+                            return Some({ f64::abs(v as f64 - solution[x]) } as f32);
+                        }
+                        _ => { None }
+                    }
+                }
+                Err(..) => {
+                    None
+                }
+            }
+        } else { None }
+    }
+}
+
+// Todo combine this with expr.is_linear() in a visitor pattern, is currently very MVP and only extracts x < 5 and such,
+// TODO is next in line to be rewritten
+fn extract_linear_expression(expr: ExprKind) -> Option<LinearExpression> {
+    match &expr {
+        ExprKind::BinaryOp(operator, operand1, operand2) => {
+            if let ExprKind::OwnedIdent(id) = &operand1.kind {
+                if let Identifier::Resolved { owner, name } = *id.clone() {
+                    let symbol_of_id = SymbolIdentifier { owner: owner.clone(), name: (name.clone()).parse().unwrap() };
+                    if let ExprKind::Number(number) = operand2.kind {
+                        match operator {
+                            Addition => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Addition }) }
+                            Multiplication => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Multiplication }) }
+                            Subtraction => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Subtraction }) }
+                            Division => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Division }) }
+                            Equality => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Equality }) }
+                            Inequality => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Inequality }) }
+                            GreaterThan => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: GreaterThan }) }
+                            LessThan => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: LessThan }) }
+                            GreaterOrEqual => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: GreaterOrEqual }) }
+                            LessOrEqual => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: LessOrEqual }) }
+                            And => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: And }) }
+                            Or => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Or }) }
+                            Xor => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Xor }) }
+                            Implication => { Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: Implication }) }
+                        }
```

Comment:
Pretty sure this could just be
```rust
Some(LinearExpression { symbol: symbol_of_id, constant: number, operation: operator })
```
There seem to be no difference between the match arms, right?

Reply:
Correct, fixed in d0e86a698edd1263f4f976d5e6a814c72d4ccb3f and e9d7146633c31d1378478c22cf39fad7c1983389